### PR TITLE
Implement some user requests and improvements

### PR DESCRIPTION
**Pull request by udjeni**
* More accurate birth and death dates (before, after, etc).
* Living people are not shown with death date "unknown" but as "Still living".
* 'Limit display' is adjusted to 'Max Gen' when the latter is changed but is re-adjusted to the max tree level after the tree has been loaded and/or before the tree is redrawn.
* Shift-click on leaf node will expand the branch to all data loaded.
* Connecters correctly displays birth and death data.
* User selected options are saved to and retrieved from a cookie.
* Negative Ids higher up the tree are now handled correctly.
* Updated help text re scrolling.

### CODE REVIEW

**Comment by udjeni** [comment]
These changes can be tested at https://apps.wikitree.com/apps/smit641/dynamic-tree/#view=ale

File: views/ancestorLines/ancestor_lines_explorer.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { AncestorTree } from "./ancestor_tree.js";
 import { showTree } from "./display.js";
 
 export class AncestorLinesExplorer {
+    static #COOKIE_NAME = "wt_ale_options";
```
Comment (MichalVasut) [inline views/ancestorLines/ancestor_lines_explorer.js:5]: Just out of curiosity, why are you using hash (#) in front of your variable names? I've never seen that. Is that some kind of javascript convention?
In UML it's for protected fields, I think, but that's just pseudo / symbolic language, never seen that in actual code. 🤔

Comment (udjeni) [reply views/ancestorLines/ancestor_lines_explorer.js:5]: Yes, in ES6 (at least, I think), it defines private variables

Comment (MichalVasut) [reply views/ancestorLines/ancestor_lines_explorer.js:5]: Cool, thanks for confirmation. 👍

**Comment by udjeni** [comment]
I found that the previous scrolling solution did not work well on Windows machines. It and dragging also did not work well together. I have now removed the draggability in favour of floating scrolling buttons. The floating scroll bars are still in place.

**Pull request merged**